src/catalog/src/durable/impls/persist.rs

```diff
@@ -291,8 +266,8 @@ impl PersistHandle {
     #[tracing::instrument(level = "debug", skip(self))]
     async fn current_snapshot(&mut self) -> (&Vec<StateUpdate>, Timestamp) {
         static EMPTY_SNAPSHOT: Vec<StateUpdate> = Vec::new();
-        let (is_initialized, current_upper) = self.is_initialized_inner().await;
-        if is_initialized {
+        let current_upper = self.current_upper().await;
+        if current_upper != Timestamp::minimum() {
```

Comment:
Why can't we use `is_initialized_inner(...)` here anymore?

Reply:
We can use `is_initialized_inner`, but it's now much more expensive. Instead of just looking at the upper of the persist shard, we actually have to read in the contents.

This has actually gotten a bit cheaper since I added caching to the `PersistHandle` today (https://github.com/MaterializeInc/materialize/pull/23444), so you don't always have to read in the shard contents. So maybe I'll bring back `is_initialized_inner`, or at least abstract this log to a method like:
```Rust
/// Reports if the persist shard can be read at some time, and the current upper.
async fn is_persist_shard_readable(&self) -> (bool, Timestamp) {
  let upper = self.current_upper().await;
  (upper > Timestamp::minumum(), upper)
}
```

Any preference between the two approaches? I think I slightly prefer the `is_persist_shard_readable` approach.

Reply:
I pushed the `is_persist_shard_readable` approach for now, but happy to switch it back if you think `is_initialized_inner` is cleaner.